src/ManipulationMsg.chpl

```diff
@@ -0,0 +1,915 @@
+        axisSizes: [d] int;
+    axisSizes[0] = shapes[0][axis];
+
+    for s in 1..<shapes.size {
+      for param i in 0..<n {
+        if i == axis {
+          shapeOut[i] += shapes[s][i];
+          axisSizes[s] = shapes[s][i];
+        } else {
+          // all non-axis dimensions must match
+          if shapes[s][i] != shapeOut[i] then
+            return (false, shapeOut, axisSizes);
+        }
+      }
+    }
+
+    const startOffsets = (+ scan axisSizes) - axisSizes;
+    return (true, shapeOut, startOffsets);
+  }
+
+  // alternative to 'concatMsg' to be used when the axis argument is 'None'
+  @arkouda.registerND
+  proc concatFlatMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, param nd: int): MsgTuple throws {
+    param pn = Reflection.getRoutineName();
+    const nArrays = msgArgs.get("n").getIntValue(),
+          names = msgArgs.get("names").getList(nArrays),
+          rname = st.nextName();
+
+    var gEnts: [0..<nArrays] borrowed GenSymEntry = getGenericEntries(names, st);
+
+    // confirm that all arrays have the same dtype
+    // (type promotion needs to be completed before calling 'concat')
+    const dt = gEnts[0].dtype;
+    for i in 1..<nArrays do if gEnts[i].dtype != dt {
+      const errMsg = "All arrays must have the same dtype to concatenate";
+      mLogger.error(getModuleName(),pn,getLineNumber(),errMsg);
+      return new MsgTuple(errMsg,MsgType.ERROR);
+    }
+
+    proc doFlatConcat(type t): MsgTuple throws {
+      const eIns = [i in 0..<nArrays] toSymEntry(gEnts[i], t, nd),
+            sizes = [i in 0..<nArrays] eIns[i].a.size,
+            starts = (+ scan sizes) - sizes;
+
+      // create a 1D output array
+      var eOut = st.addEntry(rname, + reduce sizes, t);
+
+      // copy the data from the input arrays to the output array
+      for arrIdx in 0..<nArrays {
+        const a = flatten(eIns[arrIdx].a);
+        forall idx in a.domain with (var agg = newSrcAggregator(t)) do
+          agg.copy(eOut.a[idx + starts[arrIdx]], a[idx]);
+      }
+
+      const repMsg = "created " + st.attrib(rname);
+      mLogger.info(getModuleName(),pn,getLineNumber(),repMsg);
+      return new MsgTuple(repMsg, MsgType.NORMAL);
+    }
+
+    select dt {
+      when DType.Int64 do return doFlatConcat(int);
+      when DType.UInt64 do return doFlatConcat(uint);
+      when DType.Float64 do return doFlatConcat(real);
+      when DType.Bool do return doFlatConcat(bool);
+      when DType.BigInt do return doFlatConcat(bigint);
+      otherwise {
+        var errorMsg = notImplementedError(pn,dtype2str(dt));
+        mLogger.error(getModuleName(),pn,getLineNumber(),errorMsg);
+        return new MsgTuple(errorMsg,MsgType.ERROR);
+      }
+    }
+  }
+
+  private proc flatten(const ref a: [?d] ?t): [] t throws {
+    var flat = makeDistArray({0..<d.size}, t);
+    forall idx in flat.domain with (var agg = newSrcAggregator(t)) do
+      agg.copy(flat[idx], a[d.orderToIndex(idx)]);
```

Comment:
These copy patterns are hard for me to wrap my head around... 

It seems here we are just iterating through the second array and copying the values over through aggregators into the flat array, but I am just thikning if doing more of a bulk assign where we figure out where the elements go in the flat array based on the dimensions and then do a bulk assign would be better?

e.g.:

```
flat: [ 0 0 0 0 ]
a   : [ 1 2 ; 3 4 ]
```
Since we know `a`s dimensions, we can do something like `flat[0..#flat.length/(a[0]).length)] = a[0]` 

Probably doesn't matter and no resaon to do that premature performance work before we have a reason to, but just curious what led to this pattern.